lms/djangoapps/completion/models.py

```diff
@@ -106,6 +106,86 @@ def submit_completion(self, user, course_key, block_key, completion):
+        """
+        Inserts and Updates the completion AggregateCompletion for the specified record.
+
+        Parameters:
+            * user (django.contrib.auth.models.User): The user for whom the
+              completion is being submitted.
+            * course_key (opaque_keys.edx.keys.CourseKey): The course in
+              which the submitted block is found.
+            * block_key (opaque_keys.edx.keys.UsageKey): The block that has had
+              its completion changed.
+            * aggregation_name : The name of the aggregated blocks. This is set by the level that the aggregation
+              is occuring. Possible values are "course", "chapter", "sequential", "vertical"
+            * earned: The sum of the fractional completions of all descendant blocks
+            * possible: The total sum of the possible completion values of all descendant
+              blocks that are visible to the user. This will always be a positive integer.
+
+        Return Value:
+            (BlockCompletion, bool): A tuple comprising the created or updated
+            BlockCompletion object and a boolean value indicating whether the
+            object was newly created by this call.
+
+        Raises:
+
+            ValueError:
+                If the wrong type is passed for one of the parameters.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+
+            django.db.DatabaseError:
+                If there was a problem getting, creating, or updating the
+                BlockCompletion record in the database.
+
+                This will also be a more specific error, as described here:
+                https://docs.djangoproject.com/en/1.11/ref/exceptions/#database-exceptions.
+                IntegrityError and OperationalError are relatively common
+                subclasses.
+        """
+
+        # Raise ValueError to match normal django semantics for wrong type of field.
+        if not isinstance(course_key, CourseKey):
+            raise ValueError(
+                "course_key must be an instance of `opaque_keys.edx.keys.CourseKey`.  Got {}".format(type(course_key))
+            )
+
+        percent = earned / possible
+        if percent < 0 or percent > 1:
+            raise ValueError('The calculation of percent resulted in a number less than 0 or larger than 1.')
+
+        if waffle.waffle().is_enabled(waffle.ENABLE_COMPLETION_TRACKING):
+            obj, isNew = self.get_or_create(
+                user=user,
+                course_key=course_key,
+                aggregation_name=aggregation_name,
+                block_key=block_key,
+                defaults={
+                    'percent': percent,
+                    'possible': possible,
+                    'earned': earned,
+                },
+            )
+            if not isNew and obj.percent != percent:
+                obj.percent = percent
```

Comment:
You need to update all of the fields here: `percent`, `earned`, `possible`.

I'm  not sure what the use case is for checking to see if the `percent` is changed before setting and saving (maybe speed optimization?), but `BlockCompletion` did it, so it's ok to do here too.  But you'll also need to update the check above to:
```python
if not isNew and ((obj.percent != percent) or (obj.possible != possible) or (obj.earned != earned)):
```